static/flow/views/data-set-view.js

```diff
@@ -199,7 +199,15 @@ var DataSetView = function(options) {
             displayedName = base.m_dataSet.metadata.displayedName;
         }
         info.append($('<div>').text("DataSet Name: " + displayedName));
-        info.append($('<div>').text("Program Name: " + base.m_dataSet.metadata.program.name));
+        if(base.m_dataSet.metadata.program.displayedName == null){
+            //old style dataset where name/displayed name are the same
+            //and displayedName value is null or undefined
+            info.append($('<div>').text("Program Name: " + base.m_dataSet.metadata.program.name)); 
```

Comment:
Is this preferable to just setting the `displayedName` equal to the `name` when you're updating legacy files here? https://github.com/concord-consortium/flow-server/pull/57/files#diff-3beaf466d22ed3f70c8fb6d90cd42327R90

Reply:
This is an odd case because we have a dataset which has a program saved within its metadata (because a user can run an unsaved program or modify a program that is being run, there is no secure way to grab the program saved in the server database - we actually have a copy of the program in the dataset metadata to ensure that we can access it later for operations as we see here).  In current saved datasets, `displayedName' works fine.  However, the distinction between a displayed name and an internal name is recent.  Old program representations simply used the `name` property (which referenced BOTH the internal and displayed name).  So this is the case where the program has no `displayedName` property because it was created before displayedName/name were separated.  The other branch handles the other aspect of this split - no internal name.  Internal names (a folder location) are based on the date/time when the program was created.  So, in other words:
 - programs used to just have a name.  This was shown to and editable by the user and was what we used when saving the program on the server.  Life was simple, but users could not create multiple programs with the same name.  In the classroom this caused trouble because students did not change program names and instead frequently overwrote a previous program.
 - programs now have a name and a displayed name.  The name is a unique date/time-based name that we use internally when saving the program on the server.  It represents a folder name which contains the program and program metadata.  It is not shown to the user.  The displayed name is what the user names the program.
 - old programs then have the problem that if we try to access `displayedName`, it is actually stored in `name`.  If we try to access the internal name for saving, we have one, but it does not match the current spec or format and a new one must be generated.

Reply:
OH ok thank you for this explanation, I think I get it now! The other PR deals with legacy programs, and this is dealing with legacy *datasets*, which reference legacy programs. The legacy datasets now include a `displayName` in their metadata, but it was previously just called a `name`. Does that sound correct?